src/validation.cpp

```diff
@@ -4659,7 +4662,9 @@ bool DumpMempool(void)
 }
 
 //! Guess how far we are in the verification process at the given block index
+//! needs cs_main due to access of pindex->nChainTx (mutable)
 double GuessVerificationProgress(const ChainTxData& data, const CBlockIndex *pindex) {
+    AssertLockHeld(cs_main);
```

Comment:
I think maybe the new lock and assert here could be dropped and comment above could be changed to: "require cs_main if pindex has not been validated yet (because nChainTx might be unset)."

If we're calling this on blocks from chainActive, it seems like locking should be unnecessary.

Reply:
After 3d40869, `ClientModel::getVerificationProgress` is the only caller without the lock, so `AssertLockHeld` seems fine to me.